x-pack/plugins/actions/server/builtin_action_types/cases_webhook/validators.ts

```diff
@@ -59,4 +69,59 @@ export const validateSecrets = (secrets: CasesWebhookSecretConfigurationType) =>
 export const validate: ExternalServiceValidation = {
   config: validateConfig,
   secrets: validateSecrets,
+  connector: validateConnector,
+};
+
+const validProtocols: string[] = ['http:', 'https:'];
+export const assertURL = (url: string) => {
+  try {
+    const parsedUrl = new URL(url);
+
+    if (!parsedUrl.hostname) {
+      throw new Error('URL must contain hostname');
+    }
+
+    if (!validProtocols.includes(parsedUrl.protocol)) {
+      throw new Error('Invalid protocol');
+    }
+  } catch (error) {
+    throw new Error(`${error.message}`);
+  }
+};
+export const ensureUriAllowed = (
+  url: string,
+  configurationUtilities: ActionsConfigurationUtilities
+) => {
+  try {
+    configurationUtilities.ensureUriAllowed(url);
+  } catch (allowedListError) {
+    throw Error(i18n.ALLOWED_HOSTS_ERROR(allowedListError.message));
+  }
+};
+export const normalizeURL = (url: string) => {
+  const urlWithoutTrailingSlash = url.endsWith('/') ? url.slice(0, -1) : url;
+  const replaceDoubleSlashesRegex = new RegExp('([^:]/)/+', 'g');
+  return urlWithoutTrailingSlash.replace(replaceDoubleSlashesRegex, '$1');
+};
+
+export const validateAndNormalizeUrl = (
+  url: string,
+  configurationUtilities: ActionsConfigurationUtilities,
+  urlDesc: string
+) => {
+  try {
+    assertURL(url);
+    ensureUriAllowed(url, configurationUtilities);
+    return normalizeURL(url);
+  } catch (e) {
+    throw Error(`${urlDesc} ${e}`);
+  }
+};
+
+export const validateJson = (jsonString: string, jsonDesc: string) => {
+  try {
+    JSON.parse(jsonString);
+  } catch (e) {
+    throw new Error(`JSON Error: ${jsonDesc} must be valid JSON`);
```

Comment:
I think we should not log the `jsonString` because it may contain sensitive data like in the description of a case. We should not log this kind of information for privacy reasons.